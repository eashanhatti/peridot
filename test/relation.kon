let {
	assume Man : [A : MType] -> MType;
	assume Mortal : [A : MType] -> MType;

	assume rule1 : [A : MType] -> [x : A] -> rule <Mortal x> :- <Man x>;
} in {
	(pure unit)
}
