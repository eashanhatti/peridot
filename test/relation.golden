((fromList [(Global {unGlobal = 0}, Just UnitType)],
  fromList [],
  fromList [],
  [(SourcePos
      {sourceName = "<TODO>",
       sourceLine = Pos 5,
       sourceColumn = Pos 55},
    FailedUnify
      (TypeType Meta) (FreeVar (Level {unLevel = 0}))),
   (SourcePos
      {sourceName = "<TODO>",
       sourceLine = Pos 5,
       sourceColumn = Pos 69},
    FailedUnify
      (TypeType Meta) (FreeVar (Level {unLevel = 0})))]),
 Let
   [MetaConstant
      (Id {unId = 3})
      (MetaFunType
         Explicit (TypeType Meta) (TypeType Meta)),
    MetaConstant
      (Id {unId = 7})
      (MetaFunType
         Explicit (TypeType Meta) (TypeType Meta)),
    MetaConstant
      (Id {unId = 11})
      (MetaFunType
         Explicit
         (TypeType Meta)
         (MetaFunType
            Explicit
            (LocalVar (Index {unIndex = 0}))
            (MetaFunType
               Implicit
               (MetaFunElim
                  (GlobalVar (Id {unId = 3}))
                  (LocalVar (Index {unIndex = 0})))
               (MetaFunElim
                  (GlobalVar (Id {unId = 7}))
                  (LocalVar (Index {unIndex = 0}))))))]
   (IOIntroPure UnitIntro))
